### PR TITLE
fix(groupStore): pendingChanges were not accounted for

### PR DESCRIPTION
**Pull request by evanpurkhiser**
This fixes optimistic updates of the issues stream

This was broken for who knows how long. The types were wrong.

I'll have a follow up PR that makes many of the group store types way better

### CODE REVIEW

File: static/app/stores/groupStore.tsx [hunk 1]
```diff
@@ -18,30 +18,12 @@ function showAlert(msg, type) {
   IndicatorStore.addMessage(msg, type, {duration: 4000});
 }
 
-// TODO(ts) Type this any better.
-type Change = [string, string, any];
+type ChangeId = string;
 
-class PendingChangeQueue {
```
Comment (evanpurkhiser) [inline static/app/stores/groupStore.tsx:24]: We didn't really need this class

File: static/app/stores/groupStore.tsx [hunk 2]
```diff
@@ -18,30 +18,12 @@ function showAlert(msg, type) {
...
-      change => change[0] !== changeId || change[1] !== itemId
-    );
-  }
-
-  forEach(...args: any[]) {
-    this.changes.forEach.apply(this.changes, args);
-  }
-}
+type Change = {
+  data: any;
```
Comment (evanpurkhiser) [inline static/app/stores/groupStore.tsx:24]: `// TODO(ts) Type this any better.` was for the `any` here. But a bunch of other 'changes' are typed as any too so 🤷

File: static/app/stores/groupStore.tsx [hunk 3]
```diff
@@ -18,30 +18,12 @@ function showAlert(msg, type) {
...
-    this.changes.push([changeId, itemId, data]);
-  }
-
-  remove(changeId: string, itemId?: string) {
-    this.changes = this.changes.filter(
-      change => change[0] !== changeId || change[1] !== itemId
-    );
-  }
-
-  forEach(...args: any[]) {
```
Comment (evanpurkhiser) [inline static/app/stores/groupStore.tsx:41]: This was the reason types weren't catching the problems, the forEach took any type of args

File: static/app/stores/groupStore.tsx [hunk 4]
```diff
@@ -92,13 +74,15 @@ interface GroupStoreDefinition extends StoreDefinition, InternalDefinition {
     itemIds: string[],
     response: Partial<Group>
   ) => void;
+
```
Comment (malwilley) [inline static/app/stores/groupStore.tsx:77]: Did you mean to add these empty lines here?

**Review by malwilley — APPROVED** [review]
Looks good, tested and it does do the optimistic update now.

File: static/app/stores/groupStore.tsx (hunk 4 above)
Comment (evanpurkhiser) [reply static/app/stores/groupStore.tsx:77]: yeah, it was hard to read before I think

**Pull request merged**